src/test/ensemble_tests.cpp

```diff
@@ -822,18 +827,123 @@ TEST_F(EnsembleFlowTest, PipelineDefinitionNodesWithNodeInputMissingValidation)
+    std::vector<NodeInfo> info{
+        {NodeKind::ENTRY, "request", "", std::nullopt, {{customPipelineInputName, customPipelineInputName}}},
+        {NodeKind::DL, "sum_node", "sum", std::nullopt, {{SUM_MODEL_OUTPUT_NAME, SUM_MODEL_OUTPUT_NAME}}},
+        {NodeKind::EXIT, "response"},
+    };
+
+    std::unordered_map<std::string, std::unordered_map<std::string, InputPairs>> connections;
+
+    // Missing connection for SUM_MODEL_INPUT_NAME_2
+    connections["sum_node"] = {
+        {"request", {{customPipelineInputName, SUM_MODEL_INPUT_NAME_1}}}};
+
+    connections["response"] = {
+        {"sum_node", {{SUM_MODEL_OUTPUT_NAME, customPipelineOutputName}}}};
+
+    // Create pipeline definition
+    std::unique_ptr<PipelineDefinition> pipelineDefinition = std::make_unique<PipelineDefinition>("my_new_pipeline", info, connections);
+    ASSERT_EQ(pipelineDefinition->validateNodes(manager), StatusCode::PIPELINE_NOT_ALL_INPUTS_CONNECTED);
 }
 
-TEST_F(EnsembleFlowTest, PipelineDefinitionComplexGrapgWithNoCycleValidation) {
+TEST_F(EnsembleFlowTest, PipelineDefinitionShapesNotMatchBetweenDLModelTensorsValidation) {
+    ConstructorEnabledModelManager manager;
+    ModelConfig dummy_1x10 = config;
+    ModelConfig dummy_1x20 = config;
+    dummy_1x10.setName("dummy_1x10");
+    dummy_1x20.setName("dummy_1x20");
+    dummy_1x10.setBatchSize(0);
+    dummy_1x20.setBatchSize(0);
+    ASSERT_EQ(dummy_1x10.parseShapeParameter("(1,10)"), StatusCode::OK);
+    ASSERT_EQ(dummy_1x20.parseShapeParameter("(1,20)"), StatusCode::OK);
+    ASSERT_EQ(manager.reloadModelWithVersions(dummy_1x10), StatusCode::OK);
+    ASSERT_EQ(manager.reloadModelWithVersions(dummy_1x20), StatusCode::OK);
+
+    PipelineFactory factory;
+
+    // Simulate reading from pipeline_config.json
+    std::vector<NodeInfo> info{
+        {NodeKind::ENTRY, "request", "", std::nullopt, {{customPipelineInputName, customPipelineInputName}}},
+        {NodeKind::DL, "dummy_node_1x10", "dummy_1x10", std::nullopt, {{DUMMY_MODEL_OUTPUT_NAME, DUMMY_MODEL_OUTPUT_NAME}}},
+        {NodeKind::DL, "dummy_node_1x20", "dummy_1x20", std::nullopt, {{DUMMY_MODEL_OUTPUT_NAME, DUMMY_MODEL_OUTPUT_NAME}}},
+        {NodeKind::EXIT, "response"},
+    };
+
+    std::unordered_map<std::string, std::unordered_map<std::string, InputPairs>> connections;
+
+    connections["dummy_node_1x10"] = {
+        {"request", {{customPipelineInputName, DUMMY_MODEL_INPUT_NAME}}}};
+
+    connections["dummy_node_1x20"] = {
+        {"dummy_node_1x10", {{DUMMY_MODEL_OUTPUT_NAME, DUMMY_MODEL_INPUT_NAME}}}};
+
+    connections["response"] = {
+        {"dummy_node_1x20", {{DUMMY_MODEL_OUTPUT_NAME, customPipelineOutputName}}}};
+
+    // Create pipeline definition
+    std::unique_ptr<PipelineDefinition> pipelineDefinition = std::make_unique<PipelineDefinition>("my_new_pipeline", info, connections);
+    ASSERT_EQ(pipelineDefinition->validateNodes(manager), StatusCode::INVALID_SHAPE);
+}
+
+// Disabled until CVS-36446 is done.
```

Comment:
Can we somewhat test it even without support for all precisions? I think I32/FP32 are supported so maybe use only those for test?
My fear is that this code can quickly get outdated if disabled.

Reply:
We would need to add yet another dummy model to unit test. Do we want the same model (dummy) but with other precision just for this one test?

Reply:
I am good to wait for the fix.